src/main/java/io/github/eliux/mega/DateRange.java

```diff
@@ -0,0 +1,43 @@
+package io.github.eliux.mega;
+
+import io.github.eliux.mega.error.MegaInvalidDateRangeException;
+
+import java.time.LocalDate;
+import java.time.Period;
+import java.util.Optional;
+
+/**
+ * This class can generate a range using a start and endDate
+ */
+public class DateRange {
+
+    private LocalDate startDate;
+    private LocalDate endDate;
+
+    public DateRange(LocalDate startDate, LocalDate endDate) {
+        this.startDate = Optional.ofNullable(startDate).orElseThrow(() -> new MegaInvalidDateRangeException("The start date is missiong"));
```

Comment:
Typo "missiong". Also break the line in elseThrow to make it more readable, i.e..:
```
  this.startDate = Optional.ofNullable(startDate)
                                          .orElseThrow(() -> new MegaInvalidDateRangeException("Missing start date"));
```
Otherwise the line is too big.